integrations/kujira-adapter/src/money_market.rs

```diff
@@ -0,0 +1,342 @@
+        let vault_msg =
+            receipt_vault::ExecuteMsg::Deposit(receipt_vault::DepositMsg { callback: None });
+
+        let msg = wasm_execute(contract_addr, &vault_msg, vec![asset.try_into()?])?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn withdraw(
+        &self,
+        deps: Deps,
+        contract_addr: Addr,
+        asset: Asset,
+    ) -> Result<Vec<CosmosMsg>, MoneyMarketError> {
+        let config: receipt_vault::query::ConfigResponse = deps
+            .querier
+            .query_wasm_smart(&contract_addr, &receipt_vault::query::QueryMsg::Config {})?;
+        let status: receipt_vault::query::StatusResponse = deps
+            .querier
+            .query_wasm_smart(&contract_addr, &receipt_vault::query::QueryMsg::Status {})?;
+
+        let vault_msg =
+            receipt_vault::ExecuteMsg::Withdraw(receipt_vault::WithdrawMsg { callback: None });
+
+        let msg = wasm_execute(
+            contract_addr,
+            &vault_msg,
+            coins(
+                ((Decimal::from_ratio(asset.amount, 1u128) / status.deposit_redemption_ratio)
+                    * Uint128::one())
+                .u128(),
+                config.receipt_denom,
+            ),
+        )?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn provide_collateral(
+        &self,
+        _deps: Deps,
+        contract_addr: Addr,
+        asset: Asset,
+    ) -> Result<Vec<CosmosMsg>, MoneyMarketError> {
+        let vault_msg = market::ExecuteMsg::Deposit(market::DepositMsg {
+            position_holder: None,
+        });
+
+        let msg = wasm_execute(contract_addr, &vault_msg, vec![asset.try_into()?])?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn withdraw_collateral(
+        &self,
+        _deps: Deps,
+        contract_addr: Addr,
+        asset: Asset,
+    ) -> Result<Vec<CosmosMsg>, MoneyMarketError> {
+        let vault_msg = market::ExecuteMsg::Withdraw(market::WithdrawMsg {
+            amount: asset.amount,
+            withdraw_to: None,
+        });
+
+        let msg = wasm_execute(contract_addr, &vault_msg, vec![])?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn borrow(
+        &self,
+        _deps: Deps,
+        contract_addr: Addr,
+        asset: Asset,
+    ) -> Result<Vec<CosmosMsg>, MoneyMarketError> {
+        let vault_msg = market::ExecuteMsg::Borrow(market::BorrowMsg {
+            amount: asset.amount,
+        });
+
+        let msg = wasm_execute(contract_addr, &vault_msg, vec![])?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn repay(
+        &self,
+        _deps: Deps,
+        contract_addr: Addr,
+        asset: Asset,
+    ) -> Result<Vec<CosmosMsg>, MoneyMarketError> {
+        let vault_msg = market::ExecuteMsg::Repay(market::RepayMsg {
+            position_holder: None,
+        });
+
+        let msg = wasm_execute(contract_addr, &vault_msg, vec![asset.try_into()?])?;
+
+        Ok(vec![msg.into()])
+    }
+
+    fn price(
```

Comment:
TODO: implement?

Reply:
It's already implemented in comments but these are custom messages. 
We just need to get the corresponding query